tempodb/wal/local_block.go

```diff
@@ -0,0 +1,82 @@
+package wal
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/atomic"
+
+	"github.com/grafana/tempo/tempodb/backend"
+	"github.com/grafana/tempo/tempodb/backend/local"
+	"github.com/grafana/tempo/tempodb/encoding"
+	"github.com/grafana/tempo/tempodb/encoding/common"
+	"github.com/pkg/errors"
+)
+
+// LocalBlock is a block stored in a local storage.  It can be searched and flushed to a remote backend, and
+// permanently tracks the flushed time with a special file in the block
+type LocalBlock struct {
+	encoding.BackendBlock
+	local *local.Backend
+
+	flushedTime atomic.Int64 // protecting flushedTime b/c it's accessed from the store on flush and from the ingester instance checking flush time
+}
+
+func NewLocalBlock(ctx context.Context, existingBlock *encoding.BackendBlock, l *local.Backend) (*LocalBlock, error) {
+
+	c := &LocalBlock{
+		BackendBlock: *existingBlock,
+		local:        l,
+	}
+
+	flushedBytes, err := existingBlock.Reader().Read(ctx, "flushed", c.BlockMeta().BlockID, c.BlockMeta().TenantID)
```

Comment:
I think there are two paths here that I would prefer over exposing the block Reader.

1) Providing a methods that reads/write generic files to a backend block and use that instead of exposing the reader. Something like:
```
ReadNamedObject(context, string) ([]byte, error)
WriteNamedObject(context, string, []byte) (error)
```

2) Just use `c.local.Read()` like you did with `c.local.Write()`

Reply:
Went with option 2